src/NHibernate.Test/Async/NHSpecificTest/GH1766/Fixture.cs

```diff
@@ -0,0 +1,80 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System.Collections.Generic;
+using System.Linq;
+
+using NUnit.Framework;
+using NHibernate.Linq;
+
+namespace NHibernate.Test.NHSpecificTest.GH1766
+{
+	using System.Threading.Tasks;
+	[TestFixture]
+	public class FixtureAsync : BugTestCase
+	{
+		protected override void OnSetUp()
+		{
+			using (var session = OpenSession())
+			using (var transaction = session.BeginTransaction())
+			{
+				var parent = new A
+				{
+					Items = new List<B>
+					{
+						new B(),
+						new B(),
+					},
+				};
+
+				foreach (var child in parent.Items)
+				{
+					child.A = parent;
```

Comment:
The index has to be set here, like the parent. When a list is mapped as inverse, no update occurs from it. It may at most cascade the update/save operation if cascade is enabled, but it will not set the appropriate parent and index.

Reply:
Ah, ok. So it was a misunderstanding on my side. Would it be possible to update/clarify the documentation for this use case and/or write a unit test that shows this behavior? I did expect the index column to always be set/updated, because it's an integral part of the list.

Reply:
There is also:

> The non-inverse side is used to save the in-memory representation to the database.

Anyway, a consequent update is also ongoing, awaiting review (see #1761). It notably adds some more examples and guidance, including some about lists.

I do not think we need to add more.